hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/RateLimiting.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.util;
+
+import org.apache.hadoop.classification.InterfaceAudience;
+import org.apache.hadoop.classification.InterfaceStability;
+
+/**
+ * Minimal subset of google rate limiter class.
+ * Can be used to throttle use of object stores where excess load
+ * will trigger cluster-wide throttling, backoff etc and so collapse
+ * performance.
+ */
+@InterfaceAudience.Private
+@InterfaceStability.Unstable
+public interface RateLimiting {
```

Comment:
i think we may want to go that way with the s3a client. for this committer i was trying to move it up and let us manage it here.

interesting point though, as yes, if you do it behind the FS api then the applications don't need to ask for permission themselves...but they do need to consider that some calls may block a while.

putting the feature in here what helps address the key problem we've been seeing of "many renames in job commit overloading abfs"; does nothing for any other operations.

imagine if we added the source of rate limiting to the StoreOperations, where the default limiter would be the no-op one, and abfs would return one which is shared by all users of that FS instance? this would let all spark jobs in the same VM share the same limiter. it would be hidden away in our new private interface.